qless/workers/__init__.py

```diff
@@ -76,10 +76,14 @@ def clean(cls, path):
     def sandbox(cls, path):
         '''Ensures path exists before yielding, cleans up after'''
         # Ensure the path exists and is clean
-        if not os.path.exists(path):
-            logger.debug('Making %s' % path)
+        try:
             os.makedirs(path)
-        cls.clean(path)
+            logger.debug('Making %s' % path)
+        except OSError:
+            if not os.path.isdir(path):
```

Comment:
OK, so the exception gets thrown if either the directory existed already or directory creation failed. I get it.